src/librustdoc/config.rs

```diff
@@ -417,7 +417,12 @@ impl Options {
             }
             return Err(0);
         }
+        let (lint_opts, describe_lints, lint_cap) = get_cmd_lint_options(matches, error_format);
 
+        if lint_opts.is_empty() {
+            crate::usage("rustdoc");
+            return Err(0);
+        }
```

Comment:
What does this do? Can you add a test showing the behavior?

Reply:
Hi @jyn514 !
I used this because `rustdoc` without any arguments and `rustdoc` with ambiguous arguments to an argument e.g. `rustdoc -W isrgnlk` had `matches.free` as an empty list. I looked for other params to differentiate between these two cases so that we print the default help in case no args were passed and error out in the latter case. I found that `lint_opts` was empty in case of no args and it was not empty when there was garbled output.

Do you think this is the right behavior for the two cases?

Note: Thanks a lot for the prompt review! :) I'm looking at how to add tests and keeping a check on any other feedback that you or GuillaumeGomez might have.